Update setting up XR for Godot 4.3 (#10510)
godotengine/godot-docs: tutorials/xr/setting_up_xr.rst

There are other XR related nodes and there is much more to say about these three nodes, but we'll get into that later on.

Which Renderer to use
---------------------

> Godot has 3 renderer options for projects, Compatibility, Mobile, and Forward+.

Reviewer: ```suggestion
Godot has 3 renderer options for projects: Compatibility, Mobile, and Forward+.
```

It would also be nice to link to the renderers page here too, but that can't be cherrypicked because the renderers page is 4.4 only. So let's maybe do that in a followup PR if anyone remembers

The current recommendation is to use the Mobile renderer for any desktop VR project,
and use the Compatibility renderer for any project running on a standalone headset

> like the Meta-Quest 3. XR projects will run with the Forward+ renderer, but it isn't

Reviewer: ```suggestion
like the Meta Quest 3. XR projects will run with the Forward+ renderer, but it isn't
```
This seems to be how it's usually styled?

The Vulkan implementation of OpenXR is closely integrated with Vulkan, taking over part of the Vulkan system. This requires tight integration of certain core graphics features in the Vulkan renderer which are needed before the XR system is setup. This was one of the main deciding factors to include OpenXR as a core interface.

> This also means OpenXR needs to be enabled when Godot starts in order to set things
> up correctly. The required setting can be found in your project settings under
> **XR > OpenXR**.

Reviewer: ```suggestion
This also means OpenXR needs to be enabled when Godot starts in order to set things
up correctly. Check the :ref:`Enabled<class_ProjectSettings_property_xr/openxr/enabled>`
setting in your project settings under **XR > OpenXR**.
```
I'd mention the name of the setting and link to the class reference here.

.. image:: img/openxr_settings.webp

You can find several other settings related to OpenXR here as well. These can't be
changed while your application is running. The default settings will get us started
and we will go into detail in another section of the documentation.

> You'll also need to go to **XR > Shaders** in the project settings and check the
> box to enable them. Once you've done that click the restart button.

Reviewer: ```suggestion
You'll also need to go to **XR > Shaders** in the project settings and check the
:ref:`Enabled<class_ProjectSettings_property_xr/shaders/enabled>`
box to enable them. Once you've done that click the **Save & Restart** button.
```
Link to the setting directly, also use the exact name of the button as it appears in the UI and use bold style
